zcash_client_sqlite/src/wallet/sapling.rs

```diff
@@ -135,6 +135,38 @@ fn to_spendable_note(row: &Row) -> Result<ReceivedSaplingNote<ReceivedNoteId>, S
     ))
 }
 
+// The `clippy::let_and_return` lint is explicitly allowed here because a bug in the Rust compiler
+// (https://github.com/rust-lang/rust/issues/114633) fails to identify that the `result` temporary
```

Comment:
```suggestion
// The `clippy::let_and_return` lint is explicitly allowed here because a bug in Clippy
// (https://github.com/rust-lang/rust-clippy/issues/11308) means it fails to identify that the `result` temporary
```